src/main/java/frc/robot/subsystems/hang/docs/Hang.md

```diff
@@ -0,0 +1,89 @@
+|Hang Error|logs the error of the motor (distance from wanted position)|under doSendables method|
+|Hang Current|logs the current of the motor|under doSendables method|
+|Hang Bus Voltage|logs the voltage of the canbus|under doSendables method|
+|Hang Output|logs the power output of the spark max|under doSendables method|
+| --- | --- | ---- |
+|kHangTolerance|logs the tolerance (the allowed error) of the motor|under isAtPosition method|
+
+# How To Use
+to use the subsytem, you need to create a mountable motor connected to a sparkmax/encoder
+you would also need to have the frc dashboard to be able to tweak and modify code for desired function.
+
+## Methods Needed For The Two Modes
+both Manual and Pid mode use simmialr most of the same code since the only difference between the two is that 
+manual is for testing purposes and does not use pid for control so that we can see how structualy sound the arm is.
+Pid control is primarily for in competition purposes to make the movement accurate, smooth, and to not damage the hang or motor.
+#### for instance for Manual and Pid mode, you would have to the "setupMotors"
+#### Code For "setupMotors" Method in Pid Mode:
+ //Sets motors inverted
+        m_HangMotor.setInverted(Constants.HangConstants.kMotorInverted);
+        
+        //Sets Smart Limits
+        m_HangMotor.setSmartCurrentLimit(20, Constants.HangConstants.kHangMotorLimit);
+
+        //Conversion Factors for encoders
+        m_HangEncoder.setPositionConversionFactor(Constants.HangConstants.kPositionConversionFactor);
+        m_HangEncoder.setVelocityConversionFactor(Constants.HangConstants.kVelocityConversionFactor);
+
+        m_HangMotor.setIdleMode(IdleMode.kBrake);
+
+        setPID(Constants.HangConstants.kP, Constants.HangConstants.kI, Constants.HangConstants.kD);
+        
+        m_HangMotor.burnFlash();
+notice how we initialize/log all factors for controllability of the motor.
+#### code for "setupMotors" method in Manual mode:
+ //Sets motors inverted
+        m_HangMotor.setInverted(Constants.HangConstants.kMotorInverted);
+        
+        //Sets Smart Limits
+        m_HangMotor.setSmartCurrentLimit(20, Constants.HangConstants.kHangMotorLimit);
+
+        //Conversion Factors for encoders
+        m_HangEncoder.setPositionConversionFactor(Constants.HangConstants.kPositionConversionFactor);
+        m_HangEncoder.setVelocityConversionFactor(Constants.HangConstants.kVelocityConversionFactor);
+
+        m_HangMotor.setIdleMode(IdleMode.kBrake);
+        
+        m_HangMotor.burnFlash();
+now here, notice how we remove the code that logs the "KP," "KI," and "kD," (these are variables for pid) within manual mode. we dont need to worry about not damaging the subsytem or how accurate it is, we just need to be able to move the motor in some way. the whole point of the manual mode is to see how stable the subsystem is and if it breaks or cant move, we know something is wrong and to fix it.
```

Comment:
PID is for the pid mode. setPID() sets the constants (which basically controls how aggressively the motor moves towards the target in PID mode). However, it's nice to include why you would use the manual mode, so keep that part!